packages/eslint-plugin-mdx/src/processors/markdown.ts

```diff
@@ -0,0 +1,410 @@
+   * three whitespace characters after the start offset.
+   */
+  const startOffset = getBeginningOfLineOffset(node)
+
+  /*
+   * Extract the Markdown source to determine the leading whitespace for each
+   * line.
+   */
+  const code = text.slice(startOffset, node.position.end.offset)
+  const lines = code.split('\n')
+
+  /*
+   * The parser trims leading whitespace from each line of code within the
+   * fenced code block up to the opening fence's first backtick. The first
+   * backtick's column is the AST node's starting column plus any additional
+   * indentation.
+   */
+  const baseIndent = getIndentText(text, node).length
+
+  /*
+   * Track the length of any inserted configuration comments at the beginning
+   * of the linted JS and start the JS offset lookup keys at this index.
+   */
+  const commentLength = comments.reduce(
+    (len, comment) => len + comment.length + 1,
+    0,
+  )
+
+  /*
+   * In case there are configuration comments, initialize the map so that the
+   * first lookup index is always 0. If there are no configuration comments,
+   * the lookup index will also be 0, and the lookup should always go to the
+   * last range that matches, skipping this initialization entry.
+   */
+  const rangeMap = [
+    {
+      js: 0,
+      md: 0,
+    },
+  ]
+
+  // Start the JS offset after any configuration comments.
+  let jsOffset = commentLength
+
+  /*
+   * Start the Markdown offset at the beginning of the block's first line of
+   * actual code. The first line of the block is always the opening fence, so
+   * the code begins on the second line.
+   */
+  let mdOffset = startOffset + lines[0].length + 1
+
+  /*
+   * For each line, determine how much leading whitespace was trimmed due to
+   * indentation. Increase the JS lookup offset by the length of the line
+   * post-trimming and the Markdown offset by the total line length.
+   */
+  for (let i = 0; i + 1 < lines.length; i++) {
+    const line = lines[i + 1]
+    const leadingWhitespaceLength = leadingWhitespaceRegex.exec(line)[0].length
+
+    // The parser trims leading whitespace up to the level of the opening
+    // fence, so keep any additional indentation beyond that.
+    const trimLength = Math.min(baseIndent, leadingWhitespaceLength)
+
+    rangeMap.push({
+      js: jsOffset,
+
+      // Advance `trimLength` character from the beginning of the Markdown
+      // line to the beginning of the equivalent JS line, then compute the
+      // delta.
+      md: mdOffset + trimLength - jsOffset,
+    })
+
+    // Accumulate the current line in the offsets, and don't forget the
+    // newline.
+    mdOffset += line.length + 1
+    jsOffset += line.length - trimLength + 1
+  }
+
+  return rangeMap
+}
+
+const LANGUAGES_MAPPER: Record<string, string> = {
+  javascript: 'js',
+  javascriptreact: 'jsx',
+  typescript: 'ts',
+  typescriptreact: 'tsx',
+  markdown: 'md',
+  mdown: 'md',
+  mkdn: 'md',
+}
```

Comment:
Where are you basic JS + React, and TS + React off? They don’t work:

````markdown
```text/javascript
console.log(1)
```

```javascript/react
console.log(1)
```

```javascript
console.log(1)
```

```js
console.log(1)
```

```.js
console.log(1)
```

```JS
console.log(1)
```
````

```text/javascript
console.log(1)
```

```javascript/react
console.log(1)
```

```javascript
console.log(1)
```

```js
console.log(1)
```

```.js
console.log(1)
```

```JS
console.log(1)
```

I think the `get short language` functionality seems more complex that needed? This seems closer: `(node.lang || '').toLowerCase().replace(/\./g, '')`

Reply:
It will generate a virtual filename for code blocks. What means:

```
javascript -> 0_0.javascript
```

So I add this mapper so that the virtual files will be linted correctly by ESLint again.

When I try `javascript/react` or `text/javascript`, etc in VSCode, there will be no syntax highlighting, so I don't think this is a standard usage.

Of course, I can add support for it, maybe.

Reply:
> Of course, I can add support for it, maybe.

No, I don’t think you should!

But where did you get `getShortLang` from?

It seems to do much more than needed

Reply:
@wooorm Any further advise here?

Reply:
@wooorm See also https://github.com/eslint/eslint-plugin-markdown/issues/176